### ISSUE TITLE
For model training, Pai-megatron-patch needs to consider supporting that the variables PRETRAIN_CHECKPOINT_PATH and SAVED_PRETRAIN_CHECKPOINT_PATH are in the same directory

### ISSUE THREAD

**Issue by yizhouv5** [alibaba/Pai-Megatron-Patch#608]
On two 8-card GPU nodes, using the Pai-Megatron-Patch v0.11.1 version to initiate the pre-training of the qwen3 model and enable async-save for asynchronous write distributed checkpoint, the startup parameters of megatron are as follows:
megatron_options="  \
        --save ${SAVED_PRETRAIN_CHECKPOINT_PATH} \
        --lr ${LR} \
        --min-lr ${MIN_LR} \
        --lr-decay-style cosine \
        --weight-decay 0.01 \
        --adam-beta1 0.9 \
        --adam-beta2 0.95 \
        --clip-grad 1.0 \
        --init-method-std 0.008 \
        --attention-dropout 0.0 \
        --hidden-dropout 0.0 \
        --lr-decay-iters ${LR_DECAY_ITERS} \
        --lr-warmup-iters ${LR_WARMUP_ITERS} \
        --train-iters ${TRAIN_ITERS} \
        --micro-batch-size ${BATCH_SIZE} \
        --global-batch-size ${GLOBAL_BATCH_SIZE} \
        --num-layers ${NUM_LAYERS} \
        --hidden-size ${HIDDEN_SIZE} \
        --num-attention-heads ${NUM_ATTENTION_HEADS} \
        --ffn-hidden-size ${INTERMEDIATE_SIZE} \
        --seq-length ${SEQ_LEN} \
        --max-position-embeddings ${MAX_POSITION_EMBEDDINGS} \
        --max-padding-length ${PAD_LEN} \
        --log-interval 1 \
        --log-throughput \
        --eval-interval 10000 \
        --eval-iters 10 \
        --save-interval ${SAVE_INTERVAL} \
        --tensorboard-queue-size 1 \
        --tensorboard-dir ${TENSORBOARD_DIR} \
        --log-timers-to-tensorboard \
        --log-validation-ppl-to-tensorboard \
        --tensor-model-parallel-size ${TP} \
        --pipeline-model-parallel-size ${PP} \
        --context-parallel-size ${CP} \
        --num-workers 32 \
        --extra-vocab-size ${EXTRA_VOCAB_SIZE} \
        --patch-tokenizer-type Qwen3Tokenizer \
        --swiglu \
        --normalization RMSNorm \
        --norm-epsilon ${RMS_NORM_EPS} \
        --use-rotary-position-embeddings \
        --position-embedding-type rope \
        --disable-bias-linear \
        --rotary-base ${ROPE_THETA} \
        --ckpt-format torch_dist \
        --transformer-impl transformer_engine \
        --cross-entropy-loss-fusion \
        --qk-layernorm \
        --kv-channels 128 \
        _--async-save \_
        "

Due to network or GPU failure, the training was interrupted. It is necessary to load the saved checkpoint path to continue initiating the training. The training startup script is configured as follows:
sh run_mcore_qwen3.sh  dsw  A3B   4    64 1e-5   1e-6   4096  4096  bf16  4   2  1 1 2 true true   true false false   false 200  /mnt/qwen3-datasets/mmap_qwen3_datasets_text_document   /mnt/qwen3-datasets/mmap_qwen3_datasets_text_document   /mnt/logs/output_mcore_qwen3_pretrain_20250526_01/checkpoint/pretrain-mcore-qwen3-moe-megatron-A3B-lr-1e-5-minlr-1e-6-bs-4-gbs-64-seqlen-4096-pr-bf16-tp-4-pp-2-cp-1-ac-false-do-true-sp-true-ti-381-wi-3  100000000  1000000  /mnt/logs/output_mcore_qwen3_pretrain_20250526_01

The run_mcore_qwen3.sh script loads the model directory and writes to the checkpoint directory (in addition to saving the model weights and the optimizer subdirectory, there is also the tensorboard subdirectory), which defines two variables. Only considering the scenario where these two directories are different, it is necessary to add that the processing logic of these two directories is the same. The specific modifications are as follows:

1、Original script content
find -L ${PRETRAIN_CHECKPOINT_PATH} -maxdepth 1 -type f -name "*.json" -print0 | xargs -0 cp -t ${SAVED_PRETRAIN_CHECKPOINT_PATH}
find -L ${PRETRAIN_CHECKPOINT_PATH} -maxdepth 1 -type f -name "merges.txt" -print0 | xargs -0 cp -t ${SAVED_PRETRAIN_CHECKPOINT_PATH}

2、The modified script content
if [ "$PRETRAIN_CHECKPOINT_PATH" != "$SAVED_PRETRAIN_CHECKPOINT_PATH" ]; then
    find -L ${PRETRAIN_CHECKPOINT_PATH} -maxdepth 1 -type f -name "*.json" -print0 | xargs -0 cp -t ${SAVED_PRETRAIN_CHECKPOINT_PATH}
    find -L ${PRETRAIN_CHECKPOINT_PATH} -maxdepth 1 -type f -name "merges.txt" -print0 | xargs -0 cp -t ${SAVED_PRETRAIN_CHECKPOINT_PATH}
else
    echo "Source and destination directories are the same. No copy operation needed."
fi
